File: backend/src/main/java/edu/cit/spot/entity/Course.java
package edu.cit.spot.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Table(name = "courses")
public class Course {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank
    @Size(max = 100)
    private String courseName;
    
    @Size(max = 500)
    private String courseDescription;
    
    @NotBlank
    @Size(max = 20)
    @Column(unique = true)
    private String courseCode;
    
    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<Section> sections = new ArrayList<>();
}

File: frontend_mobile/app/src/main/java/com/example/spot/model/Course.kt
package com.example.spot.model

/**
 * Course data model matching the backend CourseDto
 */
data class Course(
    val id: Long,
    val courseName: String,
    val courseDescription: String,
    val courseCode: String,
    val sectionCount: Int,
    val schedule: String? = null
)
